src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -79,21 +80,26 @@ internal sealed partial class Http2Connection : HttpConnectionBase, IDisposable
 #else
         private const int InitialConnectionBufferSize = 4096;
 #endif
-
-        private const int DefaultInitialWindowSize = 65535;
+        // According to https://httpwg.org/specs/rfc7540.html#rfc.section.6.9.2:
+        // "The connection flow-control window can only be changed using WINDOW_UPDATE frames."
+        // We need to initialize _connectionWindow with the value 65535, since
```

Comment:
This comment isn't correct.

The initial connection window size is defined in the RFC here: https://datatracker.ietf.org/doc/html/rfc7540#section-5.2.1

Reply:
Which part of the commend do you mean? My point is that the initial connection window sizes is a always 65535 according to another section (6.9.2), and can't be configured by SETTINGS frames, therefore we need a different constant.

The initial stream window size, we may want to make it configurable.

Reply:
I agree we need two constants, I just think the comment is confusing...

"higher values may violate the protocol" -- higher values are completely in violation of the protocol.

I would suggest just calling this InitialConnectionWindowSize and remove the comment entirely, or just comment with a link to https://datatracker.ietf.org/doc/html/rfc7540#section-5.2.1.